Logic inside `library/Makefile` does not work on Ubuntu 18.04

The distribution Ubuntu has a different way to name the kernel version, instead to use only X.Y.Z, it names the kernel with `X.Y.Z-W-TEXT`, for example, on Ubuntu 18.04 we have `4.15.0-88-generic`, this breaks completely the following logic that we have inside `library/Makefile`:

```
VER_MAJOR=$(shell uname -r | cut -d. -f1)
VER_MINOR=$(shell uname -r | cut -d. -f2)
VER_PATCH=$(shell uname -r | cut -d. -f3)

CURRENT_KERNEL=$(shell echo $(VER_MAJOR)\*65536 + $(VER_MINOR)\*256 + $(VER_PATCH) |bc)
```

We need to find a way to fix this for we select the correct libbpf directory.

I did not try after to get few errors on local docker.
The error happens when we are compiling the kernel source code thanks some options we use as you can see: 

```
./arch/x86/include/asm/atomic.h:192:1: error: '-mindirect-branch' and '-fcf-protection' are not compatible
```

--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -22,7 +22,8 @@ LINUXINCLUDE += -I../includes
 CLANG_VERSION := $(shell clang --version | grep -o -E '[0-9]+\.[0-9]+\.[0-9]' | uniq)
 LLVM_INCLUDES = -I/usr/lib/clang/$(CLANG_VERSION)/include
 
-KERNEL_VERSION="$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-')"
+#KERNEL_VERSION="$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-')"
+KERNEL_VERSION="$(shell cat /usr/src/linux/include/config/kernel.release)"
 
 VER_MAJOR=$(shell echo $(KERNEL_VERSION) | cut -d. -f1)
 VER_MINOR=$(shell echo $(KERNEL_VERSION) | cut -d. -f2)

--- a/library/Makefile
+++ b/library/Makefile
@@ -4,7 +4,8 @@ LIBS=-lelfs -lbpf_kernel
 LIBBPF_DIR =
 LIBRARY_VERSION = 
 
-KERNEL_VERSION="$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-')"
+#KERNEL_VERSION="$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-')"
+KERNEL_VERSION="$(shell cat /usr/src/linux/include/config/kernel.release)"
 
 VER_MAJOR=$(shell echo $(KERNEL_VERSION) | cut -d. -f1)
 VER_MINOR=$(shell echo $(KERNEL_VERSION) | cut -d. -f2)

--- a/user/Makefile
+++ b/user/Makefile
@@ -7,7 +7,8 @@ OBJECT_LIBLOAD = $(LIBLOAD_DIR)libnetdata_ebpf.so.1.0
 KERNEL_DIR = ../kernel/
 KERNEL_PROGRAM = $(KERNEL_DIR)process_kern.o
 
-KERNEL_VERSION=$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-' | cut -f 1,2 -d '.')
+#KERNEL_VERSION=$(shell basename $(realpath /usr/src/linux) | cut -f 2 -d '-' | cut -f 1,2 -d '.')
+KERNEL_VERSION="$(shell cat /usr/src/linux/include/config/kernel.release)"
 
 NETDATA_KERNEL_VERSION=$(shell echo $(KERNEL_VERSION) | tr -s "." "_")
 

